### ISSUE TITLE
[Feature Request] Mount Stamina Regen When Swimming

### ISSUE THREAD

**Issue by Deadmano** [meldurson/MountUpRestored#6]
Currently if you mount a creature and go for a swim, it will drain their stamina but once it reaches zero they will take health damage, and standing still does not regenerate their stamina. Any chance you could add an option to allow for stamina regen when idle in water, or for it to not cause damage if stamina is empty?

**Comment by meldurson** [comment]
Right now, if the mount is a water creature that is meant to avoid land then it will be able to swim totally fine. (Serpents and Leeches) The only creatures that drain stamina are land creatures that are not meant to avoid land. I can add an option to make it so land creatures don't take damage when they run out of stamina.

**Comment by Deadmano** [comment]
@meldurson I would absolutely adore that! I was horrified when I saw my wolf taking damage, and we were too far from land, stamina was not regenerating, and I am sure you can imagine what happened next!

**Comment by Deadmano** [comment]
@meldurson so I just tested the new option `Land Mounts can Drown` and while it does work mostly, there is still a case where if you sprinted (shift) and suddenly stop (at 0 stamina) the animal will continue to take tick damage and die. Also if they are at 0 stamina in the water, and you dismount and they follow, as they try to keep speed they will take tick damage and also eventually die. :( Is there a way to prevent all damage at 0 stamina, or allow for stamina regen in water for animals? This only seems to happen if they are sprinting.

**Comment by meldurson** [comment]
I will try and make it so they do not take damage if you have no stamina and set `land mounts can drown` to false

**Comment by Deadmano** [comment]
> I will try and make it so they do not take damage if you have no stamina and set `land mounts can drown` to false

It isn't when the player has no stamina, but the mount. Once they run out of stamina they take damage in water the same way players do.

**Comment by meldurson** [comment]
Yes. Sorry. What I mean is if the mounts stamina drops to 0, no matter what, if you have disabled drowning mounts it will not lose health
